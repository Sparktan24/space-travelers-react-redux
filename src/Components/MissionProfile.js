import { useSelector } from 'react-redux';
import '../styles/missionProfile.css';

const MissionProfile = () => {
  const missionsData = useSelector((store) => store.mission);
  const data = missionsData.missions;

  const reservedMissions = data.filter((item) => item.reserved === true);

  if (reservedMissions.length > 0) {
    return (
      <ul>
        {reservedMissions.map((reserved) => (
          <li key={reserved.mission_id}>{reserved.mission_name}</li>
        ))}
      </ul>
    );
  }

  return (
    <p>No missions assigned</p>
  );
};

export default MissionProfile;
