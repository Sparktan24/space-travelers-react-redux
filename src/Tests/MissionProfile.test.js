import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { useSelector } from 'react-redux';

import MissionProfile from '../Components/MissionProfile';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('MissionProfile component', () => {
  it('renders reserved missions correctly', () => {
    const mockedMissions = [
      { mission_id: 1, mission_name: 'Mission 1', reserved: true },
      { mission_id: 2, mission_name: 'Mission 2', reserved: true },
      { mission_id: 3, mission_name: 'Mission 3', reserved: false },
    ];

    useSelector.mockImplementation((selectorFn) => selectorFn(
      { mission: { missions: mockedMissions } },
    ));

    const { container, getByText } = render(<MissionProfile />);

    expect(getByText('Mission 1')).toBeInTheDocument();
    expect(getByText('Mission 2')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders "No missions assigned" when no reserved missions', () => {
    useSelector.mockImplementation((selectorFn) => selectorFn({ mission: { missions: [] } }));

    const { container, getByText } = render(<MissionProfile />);

    expect(getByText('No missions assigned')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
