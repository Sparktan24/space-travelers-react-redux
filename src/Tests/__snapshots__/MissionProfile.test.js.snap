// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionProfile component renders "No missions assigned" when no reserved missions 1`] = `
<p
  class="profile-table"
>
  No missions assigned
</p>
`;

exports[`MissionProfile component renders reserved missions correctly 1`] = `
<table
  class="profile-table"
>
  <tbody>
    <tr>
      <td>
        Mission 1
      </td>
    </tr>
    <tr>
      <td>
        Mission 2
      </td>
    </tr>
  </tbody>
</table>
`;
